src/blop/tests/test_agents.py

```diff
@@ -33,10 +35,25 @@ def test_agent(agent, RE, db):
     agent.refresh()
     agent.redigest()
 
+    # test trust domains for DOFs
+    dof = agent.dofs(active=True)[0]
+    raw_x = agent.raw_inputs(dof.name).numpy()
+    dof.trust_domain = tuple(np.nanquantile(raw_x, q=[0.2, 0.8]))
+
+    RE(agent.learn("qei", n=2))
+
+    # test trust domains for DOFs
+    obj = agent.objectives(active=True)[0]
+    raw_y = agent.raw_targets(index=obj.name).numpy()
+    obj.trust_domain = tuple(np.nanquantile(raw_y, q=[0.2, 0.8]))
+
+    RE(agent.learn("qei", n=2))
+
     # save the data, reset the agent, and get the data back
-    agent.save_data("/tmp/test_save_data.h5")
-    agent.reset()
-    agent.load_data("/tmp/test_save_data.h5")
+    if sys.version_info.minor > 9:
```

Comment:
Why is this needed?

Reply:
I see this in #100 as well actually and it had to do with numpy 2.0 being used, if that gets merged first then you can remove this check and the tests should pass.